witchcraft-metrics/src/gauge.rs

```diff
@@ -13,26 +13,15 @@
 // limitations under the License.
 use serde::Serialize;
 use serde_value::Value;
-use std::any::TypeId;
+use std::any::Any;
 use std::sync::Arc;
 
-mod private {
-    pub struct PrivacyToken;
-}
-
 /// A generalized metric which computes an arbitrary value.
 ///
 /// It is implemented for all closures returning serializable types.
-pub trait Gauge: 'static + Sync + Send {
+pub trait Gauge: Any + Sync + Send {
```

Comment:
Drive-by cleanup to minimize unsafe code in the Gauge impl now that trait upcasting works.